### PR TITLE
Restore parity between PR and Live test platforms

### PR DESCRIPTION
**Pull request by pakrym**
Live pipeline lacked NET5.0. Also normalize names to be the same.

### CODE REVIEW

**Review by AlexGhiondea — APPROVED** [review]

**Review by azabbasi — APPROVED** [review]

File: eng/pipelines/templates/stages/archetype-sdk-tests.yml [hunk 1]
```diff
@@ -52,28 +52,31 @@ parameters:
 - name: Platforms
   type: object
   default:
-    Linux_NetCore:
+    Linux:
```
Comment (weshaggard) [inline eng/pipelines/templates/stages/archetype-sdk-tests.yml:55]: Do we want to test net5.0 on linux as well or instead of windows?

Comment (pakrym) [reply eng/pipelines/templates/stages/archetype-sdk-tests.yml:55]: Maybe? I didn't want to grow/change the matrix as part of this PR. Just getting it in-sync.

Comment (pakrym) [reply eng/pipelines/templates/stages/archetype-sdk-tests.yml:55]: I'll be OK with testing NET5 on more platforms but wonder about agent use.

Comment (weshaggard) [reply eng/pipelines/templates/stages/archetype-sdk-tests.yml:55]: OK lets get it in sync for now and file an issue for adding more. We are working on more agent support but I don't think adding the linux_net5.0 will break the bank.

**Review by weshaggard — APPROVED** [review]

File: eng/pipelines/templates/stages/archetype-sdk-tests.yml (hunk 1 above)
Comment (pakrym) [reply eng/pipelines/templates/stages/archetype-sdk-tests.yml:55]: Filed https://github.com/Azure/azure-sdk-for-net/issues/17771

**Pull request merged**